includes/class-woocommerce-civicrm-manager.php

```diff
@@ -606,29 +606,46 @@ public function add_contribution( $cid, &$order ) {
 
 		$payment_instrument = $this->map_payment_instrument( $order->get_payment_method() );
 		$source = $this->generate_source( $order );
-		$params = array(
+		$params = [
 			'contact_id' => $cid,
-			'total_amount' => $rounded_subtotal,
 			// Need to be set in admin page
-			'contribution_type_id' => $default_contribution_type_id,
+			'financial_type_id' => $default_contribution_type_id,
 			'payment_instrument_id' => $payment_instrument,
-			'non_deductible_amount' => number_format( 0, 2, $decimal_separator, $thousand_separator ),
-			'fee_amount' => number_format( 0, 2, $decimal_separator, $thousand_separator ),
 			'trxn_id' => $txn_id,
 			'invoice_id' => $invoice_id,
 			'source' => $source,
 			'receive_date' => $order_paid_date,
-			'contribution_status_id' => $contribution_status_id,
+			// 'contribution_status_id' => $contribution_status_id,
 			'note' => $this->create_detail_string( $items ),
 			"$sales_tax_field_id" => $sales_tax,
 			"$shipping_cost_field_id" => $shipping_cost,
 			'campaign_id' => $campaign_name,
-		);
+		];
+
+		// the empty line items
+		$params['line_items'] = [
+			[
+				"params" => [],
+				'line_item' => []
+			]
+		];
+
+        // line item for shipping
+		if (floatval( $shipping_cost) > 0) {
+			$params['line_items'][0]['line_item'][] = array(
+				'price_field_id' => '1',
+				'qty' => 1,
+				'line_total' => $shipping_cost,
+				'unit_price' => $shipping_cost,
+				'label' => "Shipping",
+				'financial_type_id' => 8,
```

Comment:
Can we replace it with default_financial_type or a chosen financial type for recording shipping item from setting page?

Reply:
My thoughts here are that the plugin should create a new "Shipping" Financial Type in CiviCRM when enabled, if it does not already exist. And then use this "Shipping" Financial Type.

This saves having to provide yet another setting page to update and will work without any user intervention on new sites.

Reply:
@agileware-justin the plugin does have a settings page to map Financial types, enable address sync, etc. and also a setting on the Product itself, I think it does make sense to have those configurable, WooCommerce has variable shipping rates, and there's also downloadable Products (i.e. no shipping).

Reply:
@mecachisenros nice to hear from you. Cool, agree makes sense to use the existing settings (even though it's yet another settings page YASP!).